node/cn/filters/api.go

```diff
@@ -475,10 +450,29 @@ func (api *PublicFilterAPI) GetFilterChanges(id rpc.ID) (interface{}, error) {
 		f.deadline.Reset(api.timeout)
 
 		switch f.typ {
-		case PendingTransactionsSubscription, BlocksSubscription:
+		case BlocksSubscription:
 			hashes := f.hashes
 			f.hashes = nil
 			return returnHashes(hashes), nil
+		case PendingTransactionsSubscription:
+			if f.fullTx {
+				latest, err := api.backend.HeaderByNumber(context.Background(), rpc.LatestBlockNumber)
+				if err == nil && latest != nil {
+					txs := make([]map[string]interface{}, 0, len(f.txs))
+					for idx, tx := range f.txs {
+						txs = append(txs, kaiaApi.NewRPCTransaction(latest, tx, latest.Hash(), latest.Number.Uint64(), uint64(idx), chainConfig))
```

Comment:
I think `NewRPCTransaction` is misused and it's enough with `tx.MakeRPCOutput()`. The `NewRPCTransaction` assumes the tx is already mined so it needs block metadata (e.g., `index`)

Reply:
I've updated it. Thank you